assets/shared/helpers/player/videopress-adapter.js

```diff
@@ -18,6 +18,12 @@ export const EMBED_PATTERN = /(videopress|video\.wordpress)\.com\/.+/i;
  */
 export const initializePlayer = ( element, w = window ) =>
 	new Promise( ( resolve ) => {
+		// It was already initialized earlier.
+		const { duration } = element.dataset;
+		if ( duration ) {
+			resolve( element );
```

Comment:
Shouldn't we use `return resolve( element );` here instead?

In the current state, the `resolve` will be called and a new event listener will be added to the window, which may leak over time.

Here's the suggestion:

```suggestion
			return resolve( element );
```

Reply:
Good catch! I just added the return on the next line to avoid confusion with the `return resolve`. 😉